From ae796872f12f95f7e910410811d1896f34234d89 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Mon, 23 Mar 2020 10:04:07 +0200
Subject: [PATCH] [SW-10799] goya2: add bypass pcie recovery efuse

Because we have a efuse to load recovery from flash this can bypass the
ROM PCIe recovery and implement it in flash
recovery

Change-Id: I8b91f911e07b33482377e54b810b05dc97d7d267
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/efuse.h b/common/efuse.h
index b31f353..471cf39 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -17,6 +17,7 @@
 #define EFUSE_SYSCFG_OFFSET			0x3A
 #define EFUSE_SECURITY_FW_VERSION_OFFSET	0x3B
 
+#define EFUSE_SYSCFG_DISABLE_ROM_PCIE_RECOVERY	BIT(22)
 #define EFUSE_SYSCFG_DISABLE_SECONDARY_IMG	BIT(21)
 #define EFUSE_SYSCFG_BACKUP_DISABLE		BIT(20)
 #define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK	BIT(19)
diff --git a/common/pcie.c b/common/pcie.c
index 4ba5c70..5ddcc4d 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -121,5 +121,8 @@
 	security_spi_dma_mmu_bypass();
 	security_mem_cpy_mmu_bypass();
 	security_pcie_rr_config(spi_en);
-	pcie_config();
+
+	if ((arcboot_rom_global.config_fuse &
+	     EFUSE_SYSCFG_DISABLE_ROM_PCIE_RECOVERY) == 0)
+		pcie_config();
 }
